### ISSUE TITLE
Add support for EXIF/XMP licenses [EXIF OK / XMP not OK]

### ISSUE THREAD

**Issue by Wdavery** [geekitude/piwigo-plugin-simplecopyright#11]
IPTC is considered superseded by XMP for many people, and EXIF has it's own copyright field as well. Would you consider adding support for the newer standards?

I propose a simple fallback system. Look for the most "advanced" rights field, if not move to the next standard:
`xmp-dc:rights` > `exif:copyright` > `iptc:copyrightnotice`

This will ensure no matter where someone chooses to record the copyright information, it can be pulled out appropriately.

**Comment by geekitude** [comment]
Hello ?

While the idea sounds nice, I hove no image with EXIF or XMP copyright so I have no idea yet if I can easily access to that data.
Can you post a sample image here for each ? I'll check if I am able to read the data.

**Comment by Wdavery** [comment]
Certainly!
Both files have "CC BY-NC-SA 4.0" set in their respective fields, and can be read using most photo tools. For example, for myself on Linux I can read the metadata from both images in the file manager, the default photo viewer, a photo manager like DigiKam, and command line tools lke `exiftool`.

XML:
![xml](https://user-images.githubusercontent.com/12989893/193049985-852f2229-24a9-4275-a247-55b1727ced0c.jpg)

EXIF:
![exif](https://user-images.githubusercontent.com/12989893/193050548-a8b07e7d-83de-4266-8823-45cc0fc1c9b9.jpg)

**Comment by geekitude** [comment]
Hi...
It's been along time since last message so just wanted to keep you informed.

I agree that XMP sould have priority over IPTC since it is a new standard to hold same kind of data added in post.
But I must admit I struggle with retrieving XMP data. This might take a lot more time.

On the other hand, while EXIF Copyright is a recent addition, I don't think it superseeds IPTC.
Here's my point : as far as I know, EXIF Copyright is added by camera when picture is taken (let me know if I'm wrong).
If you're happy with the "default" copyright added by camera in EXIF, everything's perfect, just don't add any IPTC/XMP copyright.
If you add an IPTC/XMP copyright, it means you want to change value from EXIF.

Having different copyrights in EXIF and IPTC/XMP data sounds confusing though.

Can you detail how you manage your Copyright ? Do you have different usage for EXIF, IPTC or XMP copyrights?

**Comment by Wdavery** [comment]
You're right that a camera _can_ set the EXIF copyright, but only when the camera user has entered a value themselves. There is no "default" value as far as I am aware, it will just have no value. 

You're also right that different values in EXIF/IPTC/XMP would be confusing. This should never happen, as most software knows that these tags refer to the same thing and when you update XMP copyright for example it will update XMP, EXIF and IPTC. (Usually only if there was an existing value there)

I mainly use DigiKam to manage my photos and this is its behaviour as well. 

The only reason I suggested the fallbacks was that IPTC may not be updated in some software when a new copyright is set so it may hold onto an old value. From a programming view you would have to choose one of the three tags to use, so it makes sense to fallback to the older (EXIF) and older (IPTC) formats that are more likely to have a value if XMP is missing. 

Ultimately if XMP is proving complicated, just supporting EXIF would be a big leap in modernizing as EXIF is definitely fully supported in all photo software

**Comment by Wdavery** [comment]
But IPTC could certainly trump EXIF if thats your preference, I think a different value in each type would be exceedingly rare

**Comment by geekitude** [comment]
Thanks for those details.

I will soon add EXIF Copyright, XMP might be beyond my programing skills.
Maybe I'll add a comparison of EXIF and IPTC values and add an alert icon if there's a conflict between them.

**Comment by geekitude** [comment]
EXIF support added in version 13.a
We'll see if someone is willing to give me a hint on how to read XMP

@Wdavery : thanks for your ideas
